### DISCUSSION TITLE
How can I package walletConnect to a single js file by rollup?

### DISCUSSION THREAD

**Discussion by yvontang** [WalletConnect/walletconnect-monorepo#1232] [Developer Support]
Hello, How can I package walletConnect to a single js file by rollup?

If I use "namedExports" in rollup.config.js, it throw  a error "The namedExports option from "@rollup/plugin-commonjs" is deprecated. Named exports are now handled automatically.",
 otherwise it throw a error " 'default' is not exported by node_modules/@walletconnect/client/dist/cjs/index.js, imported by src/main.js".

Thank you!

Following is my rollup.config.js:
----------------------------------------------------------------------------------------
`import resolve from '@rollup/plugin-node-resolve'
import babel from '@rollup/plugin-babel'
import commonjs from '@rollup/plugin-commonjs'
import json from 'rollup-plugin-json';

export default {
    input: "src/main.js",
    output:{
        file: 'lib/walletConnect.js',
        format: "esm",
        name: '_wallet_connect_'
    },
    plugins: [

        resolve({
          preferBuiltins: false
        }),
  
        commonjs({
          include:"/node_modules/",
          /*namedExports: {
            'node_modules/@walletconnect/web3-provider/dist/cjs/index.js': ['WalletConnectProvider'],
          }*/
        }),
  
        babel({

          babelHelpers: 'bundled',
  
          extensions: ['.js'],
  
          presets: [
  
            '@babel/preset-env',
  
            '@babel/preset-typescript'
  
          ]
  
        }),
        json({})
      ]
}`
----------------------------------------------------------------------------------------

**Comment by xzilja** [comment]
This is something we started working on internally as well (build and dev tooling improvements), team is currently busy with finalising v2 release, but some work on this has been done in the following PR https://github.com/WalletConnect/walletconnect-monorepo/pull/1196

You might find it helpful. Note it uses esbuild instead of babel however
